src/widgets/ActivityGroups/model/hooks/useIntegrationsSync.ts

```diff
@@ -0,0 +1,21 @@
+import { actions } from "~/entities/activity/model"
+import { AppletDetailsDTO } from "~/shared/api"
+import { useAppDispatch } from "~/shared/utils"
+
+type Props = {
+  appletDetails: AppletDetailsDTO
+}
+
+const AVAILABLE_INTEGRATIONS = ["loris"]
+
+export const useIntegrationsSync = ({ appletDetails }: Props) => {
+  const dispatch = useAppDispatch()
+
+  const isLorisIntegrationAvailable = appletDetails.integrations.some(integration =>
+    AVAILABLE_INTEGRATIONS.includes(integration),
+  )
+
+  if (isLorisIntegrationAvailable) {
+    dispatch(actions.applyDataSharingSettings({ appletId: appletDetails.id }))
+  }
```

Comment:
Please use AppletIntegrationsService class. This way, you will not have to check for a specific integration.
```
useEffect(() => {
  appletIntegrationService.applyIntegrations(appletDetails);
}, [appletDetails, appletIntegrationService]);
```

Reply:
Done